modules/amm-executor/src/main/scala/org/ergoplatform/dex/executor/amm/interpreters/CFMMInterpreter.scala

```diff
@@ -1,26 +1,35 @@
 package org.ergoplatform.dex.executor.amm.interpreters
 
-import cats.FlatMap
+import cats.{Functor, Monad}
 import org.ergoplatform.ErgoLikeTransaction
-import org.ergoplatform.ergo.state.{Predicted, Traced}
-import org.ergoplatform.dex.domain.amm.CFMMPool
 import org.ergoplatform.dex.domain.amm.CFMMOrder._
+import org.ergoplatform.dex.domain.amm.CFMMOrderType.{DepositType, RedeemType, SwapType}
+import org.ergoplatform.dex.domain.amm.{CFMMOrder, CFMMPool}
+import org.ergoplatform.dex.executor.amm.interpreters.v1.InterpreterV1
+import org.ergoplatform.dex.executor.amm.interpreters.v3.InterpreterV3
 import org.ergoplatform.dex.protocol.amm.AMMType.{CFMMType, N2T_CFMM, T2T_CFMM}
-import org.ergoplatform.dex.protocol.instances._
+import org.ergoplatform.ergo.domain.Output
+import org.ergoplatform.ergo.state.{Predicted, Traced}
 import tofu.higherKind.{Mid, RepresentableK}
-import tofu.logging.Logging
+import tofu.logging.{Logging, Logs}
 import tofu.syntax.logging._
 import tofu.syntax.monadic._
 
 /** Interprets CFMM operations to a transaction
   */
 trait CFMMInterpreter[CT <: CFMMType, F[_]] {
 
-  def deposit(in: Deposit, pool: CFMMPool): F[(ErgoLikeTransaction, Traced[Predicted[CFMMPool]])]
+  def deposit(
+    in: CFMMOrder.AnyDeposit,
+    pool: CFMMPool
+  ): F[(ErgoLikeTransaction, Traced[Predicted[CFMMPool]], Output)]
 
-  def redeem(in: Redeem, pool: CFMMPool): F[(ErgoLikeTransaction, Traced[Predicted[CFMMPool]])]
+  def redeem(
+    in: CFMMOrder.AnyRedeem,
+    pool: CFMMPool
+  ): F[(ErgoLikeTransaction, Traced[Predicted[CFMMPool]], Output)]
 
-  def swap(in: SwapAny, pool: CFMMPool): F[(ErgoLikeTransaction, Traced[Predicted[CFMMPool]])]
+  def swap(in: CFMMOrder.AnySwap, pool: CFMMPool): F[(ErgoLikeTransaction, Traced[Predicted[CFMMPool]], Output)]
```

Comment:
Mixing different kinds of abstractions looks off. i.e. `Traced[Predicted[CFMMPool]]` is also `Output` at blockchain level, but here it is encoded as business entity and right after a raw `Output` is placed...